core/src/main/java/org/eclipse/hono/config/HonoConfigProperties.java

```diff
@@ -22,20 +22,24 @@
  */
 public final class HonoConfigProperties extends AbstractHonoConfig {
 
-    private static final int MIN_PAYLOAD_SIZE = 128; // bytes
+    private static final int MIN_PAYLOAD_SIZE  = 128; // bytes
+    public  static final int PORT_UNCONFIGURED = -1; // ports that are not explicitly configured
 
     private int maxInstances = 0;
     private int startupTimeout = 20;
     private boolean singleTenant = false;
     private boolean networkDebugLogging = false;
     private boolean waitForDownstreamConnection = false;
     private String bindAddress = "127.0.0.1";
-    private int port = 0;
+    private int port = PORT_UNCONFIGURED;
+    private boolean allowInsecure = false;
+    private String bindAddressInsecurePort = "127.0.0.1";
+    private int insecurePort = PORT_UNCONFIGURED;
```

Comment:
I thought that only appending "insecure" signals that e.g. the "bindingAddress" would be insecure. But this would point into the wrong direction IMHO, so I decided to use the longer phrase "InsecurePort" .
With the change of "allowInsecure" to "insecurePortEnabled" I suggest to also use this naming here. This leads to :

boolean insecurePortEnabled;
String insecurePortBindingAddress;
int insecurePort;

WDYT?